src/Execute/ExecutionEnv.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Command} from '../Backend/Command';
+import {Executor} from '../Backend/Executor';
+import {gToolchainEnvMap} from '../Toolchain/ToolchainEnv';
+
+// Specify Device with 'name' on `getEnableEnvList()` and `ExecutionEnv`.
+interface ExecutionEnv {
+  name(): string;
+  host(): string;
+  listExecutableExt(): string[];
+  isExecutionEnvAvail(name: string): boolean;
+  getEnableEnvList(): string[];
+  getInferenceCmd(model_path: string, _options?: string[]): Command;
+}
+
+class ToolchainExecutorEnv implements ExecutionEnv {
+  envName: string;  // Specified name for Each Env
+  simulator: Executor;
+  constructor(backendName: string, sim: Executor) {
+    this.envName = backendName;
+    this.simulator = sim;
+  }
+  name(): string {
+    return this.envName;
+  }
+  host(): string {
+    return 'x86';
```

Comment:
Maybe `os.platform() / process.platform()` can be used here

Reply:
Good point. This part could be replaced by https://nodejs.org/api/os.html .